drivers/spi/spi_mcux_dspi.c

```diff
@@ -110,11 +209,344 @@ static void spi_mcux_isr(const struct device *dev)
+		   (struct dma_config *)&data->tx_dma_config.dma_cfg);
+
+	return 0;
+}
+
+static int update_rx_dma(const struct device *dev)
+{
+	uint32_t rx_size = 0;
+	uint8_t *rx_buf;
+	struct spi_mcux_data *data = dev->data;
+	uint32_t frame_size_byte = (data->frame_size >> 3);
+	bool tx_only = false;
+
+	DSPI_DisableDMA(DEV_BASE(dev), (uint32_t)kDSPI_RxDmaEnable);
+	if (data->ctx.rx_len == 0) {
+		LOG_DBG("empty data no need to setup DMA");
+		return 0;
+	}
+
+	if (data->ctx.current_rx) {
+		rx_size = data->transfer_len;
+		if (data->ctx.rx_buf != NULL) {
+			rx_buf = data->ctx.rx_buf;
+		} else {
+			rx_buf = data->inner_rx_buffer->buf;
+		}
+	} else {
+		/* tx only */
+		rx_buf = data->inner_rx_buffer->buf;
+		rx_size = data->transfer_len;
+		tx_only = true;
+		LOG_DBG("tx only 0x%x, size %d", (uint32_t)rx_buf, rx_size);
+	}
+
+	if (DEV_CFG(dev)->is_dma_chn_shared) {
+		if (data->ctx.rx_len == 1) {
+			/* do not link tx on last frame*/
+			LOG_DBG("do not link tx/rx channel for last one");
+			data->rx_dma_config.dma_cfg.source_chaining_en = 0;
+			data->rx_dma_config.dma_cfg.dest_chaining_en = 0;
+		} else {
+			LOG_DBG("shared mux mode, link tx/rx channel");
+			data->rx_dma_config.dma_cfg.source_chaining_en = 1;
+			data->rx_dma_config.dma_cfg.dest_chaining_en = 1;
+			data->rx_dma_config.dma_cfg.linked_channel =
+				data->tx_dma_config.dma_channel;
+		}
+
+		data->rx_dma_block.dest_address = (uint32_t)rx_buf;
+		data->rx_dma_block.source_address =
+			DSPI_GetRxRegisterAddress(DEV_BASE(dev));
+		/* do once in share mode */
+		data->rx_dma_block.block_size = frame_size_byte;
+		data->rx_dma_config.dma_cfg.source_burst_length =
+			frame_size_byte;
+		data->rx_dma_config.dma_cfg.dest_burst_length = frame_size_byte;
+		data->rx_dma_config.dma_cfg.source_data_size = frame_size_byte;
+		data->rx_dma_config.dma_cfg.dest_data_size = frame_size_byte;
+
+	} else {
+		data->rx_dma_block.dest_address = (uint32_t)rx_buf;
+		data->rx_dma_block.source_address =
+			DSPI_GetRxRegisterAddress(DEV_BASE(dev));
+		data->rx_dma_block.block_size = rx_size;
+		data->rx_dma_config.dma_cfg.source_burst_length =
+			frame_size_byte;
+		data->rx_dma_config.dma_cfg.dest_burst_length = frame_size_byte;
+		data->rx_dma_config.dma_cfg.source_data_size = frame_size_byte;
+		data->rx_dma_config.dma_cfg.dest_data_size = frame_size_byte;
+	}
+
+	dma_config(data->rx_dma_config.dma_dev, data->rx_dma_config.dma_channel,
+		   (struct dma_config *)&data->rx_dma_config.dma_cfg);
+
+	return 0;
+}
+
+static int configure_dma(const struct device *dev)
+{
+	if (DEV_CFG(dev)->is_dma_chn_shared) {
+		LOG_DBG("shard DMA request");
+		update_tx_dma(dev);
+		update_rx_dma(dev);
+	} else {
+		update_tx_dma(dev);
+		update_rx_dma(dev);
+	}
+	return 0;
+}
+
+static void dma_callback(const struct device *dma_dev, void *callback_arg,
+			 uint32_t channel, int error_code)
+{
+	const struct device *dev = device_get_binding(callback_arg);
```

Comment:
This is a slow operation that we probably don't want to do in callback, especially if the callback is invoked in interrupt context.

Reply:
ok, revert the chagnes